### PR TITLE
webui: show inline per - step alerts and stay in the same step when configuration is invalid

### PR DESCRIPTION
**Pull request by KKoukiou**
For example when the user has an invalid disk selection step, the page looks like this when clicking 'Next':

Screen Shot 2022-03-15 at 17 45 45

### CODE REVIEW

File: ui/webui/src/components/app.jsx [hunk 1]
```diff
@@ -104,6 +108,12 @@ export const Application = () => {
         if (prevStep.prevId === "installation-destination") {
             applyDefaultStorage({ address, onAddErrorNotification, onSuccess: () => setIsStorageReady(true) });
         }
+        const stepIdx = steps.findIndex(s => s.id === stepIdReached.id);
+        const newStepIdx = steps.findIndex(s => s.id === newStep.id);
```
Comment (VladimirSlavik) [inline ui/webui/src/components/app.jsx:131]: `stepIdReached` should not use `.id` here, I think? Below you do `setStepIdReached(newStep.id);` which already does that part.

**Review by VladimirSlavik — COMMENTED** [review]
LGTM except for the one thing below...

File: ui/webui/src/components/app.jsx (hunk 1 above)
Comment (KKoukiou) [reply ui/webui/src/components/app.jsx:131]: Oh great catch!!

Comment (KKoukiou) [reply ui/webui/src/components/app.jsx]: Fixed! thanks for noticing!

**Review by VladimirSlavik — APPROVED** [review]
In that case I don't see anything else!

**Review by M4rtinK — APPROVED** [review]
Looks good to me! :)

We had some discussion about how to best handle the canJumpTo logic, but I'm find with this being improved in this PR or in a separate one. :)

**Comment by KKoukiou** [comment]
> Looks good to me! :)
> 
> We had some discussion about how to best handle the canJumpTo logic, but I'm find with this being improved in this PR or in a separate one. :)

I will send a followup.

**Pull request merged**